### ISSUE TITLE
CI Builds: AF_Error

### ISSUE THREAD

**Issue by jwarton** [arrayfire/arrayfire#474]
After installing AF and required dependencies as delineated here:
https://github.com/arrayfire/arrayfire/wiki/Build-Instructions-for-Windows

CPU and OpenCL solution configurations work without error.  However CUDAx64 is throwing the following errors:

Running with helloworld and blas example code:
128 x 128: Error in ..\src\backend\cuda\jit.cpp(315):  
CU Error (209)
..\..\..\..\src\api\cpp\blas.cpp:20: AF_ERROR 1

![helloworld_output](https://cloud.githubusercontent.com/assets/6433628/6561410/1cd4beb2-c65f-11e4-800d-f007bde641fc.jpg)

Running with pi example code:
Error in C:/jenkins-slave/workspace/arrayfire-windows/devel/src/backend/cuda/kernel/random.hpp(147):
CU Error (8): invalid device function
..\..\..\..\src\api\cpp\data.cpp:119:  AF_ERROR 1

Does anyone have suggestions or possible solutions?

**Comment by pavanky** [comment]
@jwarton Our CI builds are built for compute 3.0. You have a 2.1 card which means pre-built versions from the CI will not work. @shehzan10 Can you upload a special build for 2.0 on our CI ?

**Comment by jwarton** [comment]
Shehzan prepared a 2.0 build for use with my card.  Build 217.  This is the build I am currently using.  The prebuilt dependencies are however from the same .dll and .lib files that were downloaded when I initially installed AF.

Let me know if there is another build I should be using.

Thanks,

James Warton
Research Assistant, PhD Candidate
Research Center for Advanced Manufacturing
Lyle School of Engineering | SMU
codeidetic.net<http://codeidetic.net/>

**Comment by shehzan10** [comment]
So I set the options for the CI build to be 2.0. I'm trying to verify this and will get back to you.

**Comment by shehzan10** [comment]
I've verified that the build was with 2.0 (you can check build/src/backend/cuda for any ptx files and all the headers target sm_20.
Can you remind me why you weren't able to build it using the batch scripts?

**Comment by jwarton** [comment]
There are .ptx files in the directory you mentioned below.  I’m not sure what I should be looking for in the headers files, but I did not find any reference to ‘sm_20’

I never tried to use cmake or build arrayfire myself.  This isn’t a procedure I’m very familiar with.  Most of the work I’ve done depends on prebuilt libraries.  If you think this is the only way to get around the issue I’m having, you’ll have to walk me through that procedure.

Best,

James Warton
Research Assistant, PhD Candidate
Research Center for Advanced Manufacturing
Lyle School of Engineering | SMU
codeidetic.net<http://codeidetic.net/>

**Comment by shehzan10** [comment]
The top of the tile contains the ptx header, which has the line `target sm_20`.

I believe building it with the scripts will most likely fix this issue. You will not need to interact with CMake in all likelihood. 

The scripts and instructions for building ArrayFire can be found here https://github.com/shehzan10/arrayfire-windows-scripts.

Let me know if you run into any issues.

**Comment by jwarton** [comment]
I’ll look at the build procedure tonight when I get home from the lab.
None of the .hpp files within …/cuda/ptx_header contain the line target sm_20.

James Warton
Research Assistant, PhD Candidate
Research Center for Advanced Manufacturing
Lyle School of Engineering | SMU
codeidetic.net<http://codeidetic.net/>

**Comment by shehzan10** [comment]
I didn't mean hpp files, I meant the header (the first 3-4 lines) of the ptx files.

**Comment by jwarton** [comment]
They contain this:


// Generated by NVIDIA NVVM Compiler
// Compiler built on Fri Jul 25 07:36:16 2014 (1406288176)
// Cuda compilation tools, release 6.5, V6.5.13
//

.version 4.1
.target sm_20
.address_size 64



James Warton
Research Assistant, PhD Candidate
Research Center for Advanced Manufacturing
Lyle School of Engineering | SMU
codeidetic.net<http://codeidetic.net/>

**Comment by shehzan10** [comment]
Right, as you can see it says `target sm_20` which means it built correctly on the CI machine.
This should have worked on your system, so I'm still unsure as to why it did not.

**Comment by jwarton** [comment]
I’m trying to follow the instructions posted for building ArrayFire, but I’m a little confused.

I created a new directory as described and have copied the required .bat files to this location.
Revised common.bat with the path to where all .bat files are located.
Extracted the arrayfire_deps.zip into this location

I did not clone ArrayFire or Boost.Compute.  Do I need to follow this step if I plan to use CUDA?


If this is all that’s required, I’m ready to run the rebuild_arrayfire.bat.

Thanks,

James Warton
Research Assistant, PhD Candidate
Research Center for Advanced Manufacturing
Lyle School of Engineering | SMU
codeidetic.net<http://codeidetic.net/>
